packages/graphiql/src/components/QueryEditor.tsx

```diff
@@ -83,210 +108,172 @@ export class QueryEditor extends React.Component<QueryEditorProps, {}>
+          }),
         'Alt-Space': () =>
           // @ts-ignore showHint method needs improvement on definatelytyped
-          editor.showHint({ completeSingle: true, container: this._node }),
+          editorRef.current.showHint({
+            completeSingle: true,
+            // @ts-ignore
+
+            container: editorEl,
+          }),
         'Shift-Space': () =>
           // @ts-ignore showHint method needs improvement on definatelytyped
-          editor.showHint({ completeSingle: true, container: this._node }),
+          editorRef.current.showHint({
+            completeSingle: true,
+            // @ts-ignore
+
+            container: editorEl,
+          }),
         'Shift-Alt-Space': () =>
           // @ts-ignore showHint method needs improvement on definatelytyped
-          editor.showHint({ completeSingle: true, container: this._node }),
+          editorRef.current.showHint({
+            completeSingle: true,
+            // @ts-ignore
 
-        'Cmd-Enter': () => {
-          if (this.props.onRunQuery) {
-            this.props.onRunQuery();
-          }
-        },
+            container: editorEl,
+          }),
+
+        'Cmd-Enter': () => session.executeOperation(),
         'Ctrl-Enter': () => {
-          if (this.props.onRunQuery) {
-            this.props.onRunQuery();
-          }
+          console.log(session.operation.text);
+          session.executeOperation();
         },
 
         'Shift-Ctrl-C': () => {
-          if (this.props.onCopyQuery) {
-            this.props.onCopyQuery();
+          if (props.onCopyQuery) {
+            props.onCopyQuery();
           }
         },
 
         'Shift-Ctrl-P': () => {
-          if (this.props.onPrettifyQuery) {
-            this.props.onPrettifyQuery();
+          if (props.onPrettifyQuery) {
+            props.onPrettifyQuery();
           }
         },
 
         /* Shift-Ctrl-P is hard coded in Firefox for private browsing so adding an alternative to Pretiffy */
-
         'Shift-Ctrl-F': () => {
-          if (this.props.onPrettifyQuery) {
-            this.props.onPrettifyQuery();
+          if (props.onPrettifyQuery) {
+            props.onPrettifyQuery();
           }
         },
 
         'Shift-Ctrl-M': () => {
-          if (this.props.onMergeQuery) {
-            this.props.onMergeQuery();
+          if (props.onMergeQuery) {
+            props.onMergeQuery();
           }
         },
         ...commonKeys,
-        'Cmd-S': () => {
-          if (this.props.onRunQuery) {
-            // empty
-          }
-        },
-
-        'Ctrl-S': () => {
-          if (this.props.onRunQuery) {
-            // empty
-          }
-        },
+        'Cmd-S': () => {},
+        'Ctrl-S': () => {},
       },
-    }));
-    if (editor) {
-      editor.on('change', this._onEdit);
-      editor.on('keyup', this._onKeyUp);
+    });
+
+    const editHandler = () => _onEdit(session.changeOperation);
+    if (editorRef.current) {
```

Comment:
This should always be true, given that editorRef.current was assigned in the section above.

Reply:
in the case of SSR it would not be available yet! thats why a lot of these checks exist before and after this PR

Reply:
Ah. Makes sense.

Reply:
On that topic, is `useEffect` actually executed when using ssr?

Reply:
ah good point, nope. if you can make a PR that keeps typescript happy here then be my guest!

Reply:
I was messing around with a neat little 404 page for our site (https://stitch.money/404/) that used the canvas and needed to dynamically resize the canvas based on screen size, and while I could be wrong, I don't think useEffect was being called when being rendered by gatsby.

Reply:
More evidence: https://github.com/gatsbyjs/gatsby/issues/13947#issuecomment-491214724

Reply:
this all makes so much sense now. I knew this so well for the old component lifecycle, haha!